# List Comprehensions Lab

Complete the following set of exercises to solidify your knowledge of list comprehensions.

In [3]:
import os
import numpy as np
import pandas as pd

### 1. Use a list comprehension to create and print a list of consecutive integers starting with 1 and ending with 50.

In [10]:
lista_1_50 = [i for i in range(1,51)]
print(lista_1_50)

[1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50]


### 2. Use a list comprehension to create and print a list of even numbers starting with 2 and ending with 200.

In [9]:
lista_2_200 = [i for i in range(1,201) if i%2==0]
print(lista_2_200)

[2, 4, 6, 8, 10, 12, 14, 16, 18, 20, 22, 24, 26, 28, 30, 32, 34, 36, 38, 40, 42, 44, 46, 48, 50, 52, 54, 56, 58, 60, 62, 64, 66, 68, 70, 72, 74, 76, 78, 80, 82, 84, 86, 88, 90, 92, 94, 96, 98, 100, 102, 104, 106, 108, 110, 112, 114, 116, 118, 120, 122, 124, 126, 128, 130, 132, 134, 136, 138, 140, 142, 144, 146, 148, 150, 152, 154, 156, 158, 160, 162, 164, 166, 168, 170, 172, 174, 176, 178, 180, 182, 184, 186, 188, 190, 192, 194, 196, 198, 200]


### 3. Use a list comprehension to create and print a list containing all elements of the 10 x 4 Numpy array below.

In [11]:
a = np.array([[0.84062117, 0.48006452, 0.7876326 , 0.77109654],
       [0.44409793, 0.09014516, 0.81835917, 0.87645456],
       [0.7066597 , 0.09610873, 0.41247947, 0.57433389],
       [0.29960807, 0.42315023, 0.34452557, 0.4751035 ],
       [0.17003563, 0.46843998, 0.92796258, 0.69814654],
       [0.41290051, 0.19561071, 0.16284783, 0.97016248],
       [0.71725408, 0.87702738, 0.31244595, 0.76615487],
       [0.20754036, 0.57871812, 0.07214068, 0.40356048],
       [0.12149553, 0.53222417, 0.9976855 , 0.12536346],
       [0.80930099, 0.50962849, 0.94555126, 0.33364763]])

In [26]:
lista_a =[j for i in a for j in i]

print(lista_a)

[0.84062117, 0.48006452, 0.7876326, 0.77109654, 0.44409793, 0.09014516, 0.81835917, 0.87645456, 0.7066597, 0.09610873, 0.41247947, 0.57433389, 0.29960807, 0.42315023, 0.34452557, 0.4751035, 0.17003563, 0.46843998, 0.92796258, 0.69814654, 0.41290051, 0.19561071, 0.16284783, 0.97016248, 0.71725408, 0.87702738, 0.31244595, 0.76615487, 0.20754036, 0.57871812, 0.07214068, 0.40356048, 0.12149553, 0.53222417, 0.9976855, 0.12536346, 0.80930099, 0.50962849, 0.94555126, 0.33364763]


### 4. Add a condition to the list comprehension above so that only values greater than or equal to 0.5 are printed.

In [25]:
lista_a_mayor05 = [j for i in a for j in i if j>0.5]

print(lista_a_mayor05)

[0.84062117, 0.7876326, 0.77109654, 0.81835917, 0.87645456, 0.7066597, 0.57433389, 0.92796258, 0.69814654, 0.97016248, 0.71725408, 0.87702738, 0.76615487, 0.57871812, 0.53222417, 0.9976855, 0.80930099, 0.50962849, 0.94555126]


### 5. Use a list comprehension to create and print a list containing all elements of the 5 x 2 x 3 Numpy array below.

In [23]:
b = np.array([[[0.55867166, 0.06210792, 0.08147297],
        [0.82579068, 0.91512478, 0.06833034]],

       [[0.05440634, 0.65857693, 0.30296619],
        [0.06769833, 0.96031863, 0.51293743]],

       [[0.09143215, 0.71893382, 0.45850679],
        [0.58256464, 0.59005654, 0.56266457]],

       [[0.71600294, 0.87392666, 0.11434044],
        [0.8694668 , 0.65669313, 0.10708681]],

       [[0.07529684, 0.46470767, 0.47984544],
        [0.65368638, 0.14901286, 0.23760688]]])

In [33]:
c=[]
for i in b:
    for j in i:
        for k in j:
            c.append(k)

print("con condicionales = ",c)        

lista_b = [k for i in b for j in i for k in j]
print("Lista_b =", lista_b)

con condicionales =  [0.55867166, 0.06210792, 0.08147297, 0.82579068, 0.91512478, 0.06833034, 0.05440634, 0.65857693, 0.30296619, 0.06769833, 0.96031863, 0.51293743, 0.09143215, 0.71893382, 0.45850679, 0.58256464, 0.59005654, 0.56266457, 0.71600294, 0.87392666, 0.11434044, 0.8694668, 0.65669313, 0.10708681, 0.07529684, 0.46470767, 0.47984544, 0.65368638, 0.14901286, 0.23760688]
Lista_b = [0.55867166, 0.06210792, 0.08147297, 0.82579068, 0.91512478, 0.06833034, 0.05440634, 0.65857693, 0.30296619, 0.06769833, 0.96031863, 0.51293743, 0.09143215, 0.71893382, 0.45850679, 0.58256464, 0.59005654, 0.56266457, 0.71600294, 0.87392666, 0.11434044, 0.8694668, 0.65669313, 0.10708681, 0.07529684, 0.46470767, 0.47984544, 0.65368638, 0.14901286, 0.23760688]


### 5. Add a condition to the list comprehension above so that the last value in each subarray is printed, but only if it is less than or equal to 0.5.

In [69]:
d=[]
for i in b:
    for j in i:
        if j[2] >= 0.5:
            d.append(j[2])
print(d)

lista_b05 = [j[2] for i in b for j in i if j[2]>=0.5]
print(lista_b05)

[0.51293743, 0.56266457]
[0.51293743, 0.56266457]


### 6. Use a list comprehension to select and print the names of all CSV files in the */data* directory.

In [81]:
import os

listacsv = []
with open ("/Users/rube/datamex_082020/module-1/lab-list-comprehensions/data/sample_file_0.csv","r") as f:
    lines= f.readlines()
    listacsv.append(lines)
    
print(listacsv)

[['0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19\n', '0.7347510852128797,0.195361547414176,0.7343092314001137,0.598184150790113,0.7634334207035142,0.26343379232316166,0.8680663549990865,0.05809249997873411,0.7535023641545986,0.5875129985121497,0.31160828188427525,0.17835553601357446,0.18292212584258682,0.14763098677762876,0.39118785747979345,0.8160493911980281,0.7490680818183477,0.29326038172884805,0.9378284594731399,0.8808575048234396\n', '0.7726065618884856,0.44539068706313334,0.24964158612415766,0.7879222598307901,0.598582600039443,0.8272382506031565,0.6241257222313511,0.6015241437645209,0.688753283794671,0.33887040208369323,0.08159537841871223,0.471474408740446,0.26744280343306903,0.45335093437290996,0.8007161479739965,0.04574912846779866,0.6837931288231841,0.3897887717569518,0.01678665351522224,0.5036946763567658\n', '0.2264277057394144,0.2687636255734599,0.6942620171804784,0.6223345010570517,0.06384346046750078,0.12268334759663768,0.815624834853178,0.5845421568821125,0.032594

In [82]:
lista_cv
with open (os.listdir(/Users/rube/datamex_082020/module-1/lab-list-comprehensions/data/), "r") as f:
    lista_cv.append(lines)
    

SyntaxError: invalid syntax (<ipython-input-82-61becc2ca222>, line 2)

### 7. Use a list comprehension and the Pandas `read_csv` and `concat` methods to read all CSV files in the */data* directory and combine them into a single data frame. Display the top 10 rows of the resulting data frame.

### 8. Use a list comprehension to select and print the column numbers for columns from the data set whose median is less than 0.48.

### 9. Use a list comprehension to add a new column (20) to the data frame whose values are the values in column 19 minus 0.1. Display the top 10 rows of the resulting data frame.

### 10. Use a list comprehension to extract and print all values from the data set that are between 0.7 and 0.75.